### PR TITLE
chore(wal): update tests to run in wal enabled mode

### PR DESCRIPTION
**Pull request by glasstiger**
Enabled UPDATE tests for WAL tables.
PR contains 4 fixes:
- do not use JIT for filters in UPDATE statements when compiled for WAL. There is no point as it will not be executed, it is saved into the WAL as a SQL event
- pass the null flag of symbol maps in data WAL events
- MetadataValidator rejected geohash types because their column type id is greater than ColumType.MAX
- apply job to close alter/update operations

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 23 / 23 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriterEvents.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/griffin/UpdateOperatorImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalEventCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/SymbolMapDiffImpl.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 23 / 23 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriterEvents.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/griffin/UpdateOperatorImpl.java|3|3|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalEventCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/SymbolMapDiffImpl.java|2|2|100.00%|

File: core/src/main/java/io/questdb/griffin/UpdateOperatorImpl.java [hunk 1]
```diff
@@ -244,7 +244,11 @@ public long executeUpdate(SqlExecutionContext sqlExecutionContext, UpdateOperati
             tableWriter.rollbackUpdate();
             throw th;
         } finally {
-            op.closeWriter();
+            if (sqlExecutionContext.isWalApplication()) {
```
Comment (ideoma) [inline core/src/main/java/io/questdb/griffin/UpdateOperatorImpl.java:247]: I think here it should call `closeWriter()` unconditionally as before.
To fix resource leak, `ApplyWal2TableJob` should close the UpdateOperation.

Comment (glasstiger) [reply core/src/main/java/io/questdb/griffin/UpdateOperatorImpl.java:247]: best would be to remove async mode completely :)
will have a look at closing it from apply job.

Comment (glasstiger) [reply core/src/main/java/io/questdb/griffin/UpdateOperatorImpl.java]: changed it

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 27 / 27 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriterEvents.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/SymbolMapWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalEventCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/SymbolMapDiffImpl.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 27 / 27 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriterEvents.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/SymbolMapWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalEventCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/SymbolMapDiffImpl.java|2|2|100.00%|

**Review by ideoma — DISMISSED** [review]

**Review by ideoma — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 30 / 30 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriterEvents.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalWriter.java|10|10|100.00%|
|:large_blue_circle:|io/questdb/cairo/SymbolMapWriter.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/ApplyWal2TableJob.java|6|6|100.00%|
|:large_blue_circle:|io/questdb/cairo/TableWriter.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/WalEventCursor.java|2|2|100.00%|
|:large_blue_circle:|io/questdb/cairo/wal/SymbolMapDiffImpl.java|5|5|100.00%|

**Pull request merged**